Allow functionalization to work with optional mutable

flaviotruzzi opened the pull request:
Summary: - Added functionalization to allow Optionals

Test Plan: CI tests.

Reviewed By: zou3519

Differential Revision: D51209981

zou3519 commented on test/test_python_dispatch.py:
```diff
@@ -466,7 +466,7 @@ def _check_is_functional_variant(self, mutable_op, functional_op, args):
         # check rest of functional_result is the mutated args
         mutated_args = [maybe_mutated_arg for maybe_mutated_arg, arg in zip(cloned_args, args)
-                        if not torch.allclose(maybe_mutated_arg, arg)]
+                        if not(maybe_mutated_arg is not None and arg is not None and torch.allclose(maybe_mutated_arg, arg))]
```
I'm not sure this is going to pass lint (might need a space between `not` and the parenthesis

flaviotruzzi replied:
It passed :)

flaviotruzzi committed:
Allow functionalization to work with optional mutable (#114803)

Summary:

- Added functionalization to allow Optionals

Test Plan: CI tests.

Reviewed By: zou3519

Differential Revision: D51209981
